doc/development.md

```diff
@@ -28,24 +28,23 @@ Install additional packages for Fedora, CentOS, and RHEL:
 
     sudo dnf install -y `cat automation/check-patch.packages`
 
-Create virtual environment for vdsm:
-
-    make venv
-
 
 ## Building Vdsm
 
-To configure sources (run `./configure --help` to see configuration options):
+To prepare sources (create Makefiles and configure script):
 
-    git clean -xfd
     ./autogen.sh --system --enable-timestamp
-    make
 
-To test Vdsm (refer to tests/README for further tests information):
+Now you can create the virtual environment (https://docs.python.org/3/library/venv.html),
+which is necessary to run the tests later. This needs to be done only once:
+
+    make venv
```

Comment:
I don't understand what follows from that? Is my wording incorrect or missing something? Or is this the wrong place to put it?

Reply:
I mean that we don't need this here, it should be in the "Enviroment setup" section where it was before.

Reply:
The reason it was moved away from there was that it isn't possible to do it before the Makefile is created. It's confusing for the reader who follows instructions from top to bottom if a command just fails because the instructions give the commands in an order in which they can't be executed.

Reply:
This is why we need to add .autogen.sh to "Enviroment setup":

- add custom repos
- install development tools
- clone the git repo
- autogen.sh
- create the env

What's wrong with this flow?

Then in the section about building vdsm we can have:

- autogen.sh (should not be needed, but currently required)
- build the rpms
- install the rpms

And in running the tests:

- enter the env
- run the tests

Reply:
Nothing wrong with that except maybe for the duplication of the autogen. But I think we can't avoid all the little issues, so I do it like that now.